File: ios/Classes/Utils/AdDiagnosticsHelper.swift
import Foundation
import BUAdSDK

enum AdDiagnosticsHelper {
    static func map(_ list: [BUMAdLoadInfo]?, adType: String) -> [[String: Any]] {
        return list?.map { info in
            var item: [String: Any] = [
                "mediationRit": info.mediationRit,
                "adnName": info.adnName,
                "adType": adType,
                "errorCode": info.errCode,
                "errorMessage": info.errMsg
            ]
            if let customAdnName = info.customAdnName {
                item["customAdnName"] = customAdnName
            }
            if let errorUserInfo = info.errUserInfo {
                item["errorUserInfo"] = errorUserInfo
            }
            return item
        } ?? []
    }
}

File: android/src/main/kotlin/com/zhecent/gromore_ads_kit/utils/AdDiagnosticsHelper.kt
package com.zhecent.gromore_ads_kit.utils

import com.bytedance.sdk.openadsdk.mediation.manager.MediationBaseManager

object AdDiagnosticsHelper {
    fun getAdLoadInfo(manager: MediationBaseManager?): List<Map<String, Any>> {
        return manager?.adLoadInfo?.map { info ->
            mapOf(
                "mediationRit" to (info.mediationRit ?: ""),
                "adnName" to (info.adnName ?: ""),
                "adType" to (info.adType ?: ""),
                "errorCode" to info.errCode,
                "errorMessage" to (info.errMsg ?: "")
            )
        } ?: emptyList()
    }
}
